File: service/auth/src/main/java/jeonghyeon/msa/auth/exception/advice/ExceptionAdvice.java
package jeonghyeon.msa.auth.exception.advice;

import jeonghyeon.msa.auth.exception.ErrorResult;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@Slf4j
@RestControllerAdvice
public class ExceptionAdvice {

    @ResponseStatus(HttpStatus.BAD_REQUEST)
    @ExceptionHandler(IllegalArgumentException.class)
    public ErrorResult illegalState(IllegalArgumentException e) {
        log.error(e.getMessage());
        return new ErrorResult(e.getMessage());
    }
}

File: service/memo-service/src/main/kotlin/jeonghyeon/msa/memo/exception/advice/ExceptionAdvice.kt
package jeonghyeon.msa.memo.exception.advice

import jeonghyeon.msa.memo.exception.ErrorResult
import lombok.extern.slf4j.Slf4j
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.bind.annotation.RestControllerAdvice

@Slf4j
@RestControllerAdvice
class ExceptionAdvice {
    @ResponseStatus(HttpStatus.BAD_REQUEST)
    @ExceptionHandler(
        IllegalArgumentException::class
    )
    fun illegalState(e: IllegalArgumentException): ErrorResult {

        return ErrorResult(e.message)
    }
}